### PR TITLE
fix load time for ELF

### PR DESCRIPTION
**Pull request by alvarofe**
Though i found other bottleneck. `r_anal_build_range_on_hints` is so slow and is my fault when I added arm/thumb doing impossible to use radare with arm with a huge number of symbols will look at it.

### CODE REVIEW

File: libr/bin/format/elf/elf.c [hunk 1]
```diff
@@ -2161,20 +2199,28 @@ RBinElfSymbol* Elf_(r_bin_elf_get_symbols)(struct Elf_(r_bin_elf_obj_t) *bin, in
...
 	ut32 size = 0;
-	RBinElfSymbol  *ret = NULL;
 	Elf_(Shdr) *strtab_section = NULL;
 	Elf_(Sym) *sym = NULL;
 	char *strtab = NULL;
 
-	if (!bin || !bin->shdr || bin->ehdr.e_shnum == 0 || bin->ehdr.e_shnum == 0xffff)
+	if (g_symbols && type == R_BIN_ELF_SYMBOLS) {
+		return g_symbols;
+	} else if (g_imports && type == R_BIN_ELF_IMPORTS) {
```
Comment (radare) [inline libr/bin/format/elf/elf.c]: else is unnecessary

**Comment by Maijin** [comment]
Regressions

**Comment by radare** [comment]
a lot of them

**Comment by radare** [comment]
ping

**Comment by radare** [comment]
please rebase

**Comment by alvarofe** [comment]
pong, i'm back

File: libr/bin/format/elf/elf.c [hunk 2]
```diff
@@ -2352,6 +2357,23 @@ RBinElfSymbol* Elf_(r_bin_elf_get_symbols)(struct Elf_(r_bin_elf_obj_t) *bin, in
...
 }
 
+RBinElfSymbol *Elf_(r_bin_elf_get_symbols)(struct Elf_(r_bin_elf_obj_t) *bin) {
+	if (bin->g_symbols) {
+		return bin->g_symbols;
+	}
+	bin->g_symbols = Elf_(_r_bin_elf_get_symbols_imports) (bin, R_BIN_ELF_SYMBOLS);
+	return bin->g_symbols;
+}
+
```
Comment (radare) [inline libr/bin/format/elf/elf.c]: empty line

**Pull request closed without merging**